mayastor/src/persistent_store.rs

```diff
@@ -0,0 +1,250 @@
+                        // flooding the logs.
+                        error!(
+                            "Failed to connect to etcd on endpoint {}",
+                            endpoint
+                        );
+                        output_err = false;
+                    }
+                    tokio::time::sleep(Duration::from_secs(1)).await;
+                }
+            }
+        }
+    }
+
+    /// Put a key-value in the store.
+    pub async fn put(
+        key: &impl StoreKey,
+        value: &impl StoreValue,
+    ) -> Result<(), StoreError> {
+        let put_value = serde_json::to_value(value)
+            .expect("Failed to convert value to a serde_json value");
+        let key_string = key.to_string();
+        let value_clone = put_value.clone();
+
+        let rx = Self::execute_store_op(async move {
+            info!(
+                "Putting key {}, value {} in store.",
+                key_string,
+                value_clone.to_string()
+            );
+            match Self::backing_store()
+                .put_kv(&key_string, &value_clone)
+                .await
+            {
+                Ok(_) => {
+                    info!(
+                        "Successfully put key {}, value {} in store.",
+                        key_string,
+                        value_clone.to_string()
+                    );
+                    Ok(None)
+                }
+                Err(e) => Err(e),
+            }
+        });
+
+        let result = rx.await.context(PutWait {
+            key: key.to_string(),
+            value: put_value.to_string(),
+        })?;
+        result.map(|_| ())
+    }
+
+    /// Retrieve a value, with the given key, from the store.
+    pub async fn get(key: &impl StoreKey) -> Result<Value, StoreError> {
+        let key_string = key.to_string();
+        let rx = Self::execute_store_op(async move {
+            info!("Getting key {} from store.", key_string);
+            match Self::backing_store().get_kv(&key_string).await {
+                Ok(value) => {
+                    info!("Successfully got key {}", key_string);
+                    Ok(Some(value))
+                }
+                Err(e) => Err(e),
+            }
+        });
+        rx.await
+            .context(GetWait {
+                key: key.to_string(),
+            })?
+            .map(|value| value.unwrap())
+    }
+
+    /// Delete the entry in the store with the given key.
+    pub async fn delete(key: &impl StoreKey) -> Result<(), StoreError> {
+        let key_string = key.to_string();
+        let rx = Self::execute_store_op(async move {
+            info!("Deleting key {} from store.", key_string);
+            match Self::backing_store().delete_kv(&key_string).await {
+                Ok(_) => {
+                    info!(
+                        "Successfully deleted key {} from store.",
+                        key_string
+                    );
+                    Ok(None)
```

Comment:
similarly, could this be Ok(())?

Reply:
As discussed, changed the execute store op function to be generic in its return type.